lightning-background-processor/src/lib.rs

```diff
@@ -273,21 +289,38 @@ impl BackgroundProcessor {
 				// continuing our normal cadence.
 				if last_prune_call.elapsed().as_secs() > if have_pruned { NETWORK_PRUNE_TIMER } else { FIRST_NETWORK_PRUNE_TIMER } {
 					if let Some(ref handler) = net_graph_msg_handler {
-						log_trace!(logger, "Pruning network graph of stale entries");
-						handler.network_graph().remove_stale_channels();
-						if let Err(e) = persister.persist_graph(handler.network_graph()) {
-							log_error!(logger, "Error: Failed to persist network graph, check your disk and permissions {}", e)
+						log_trace!(logger, "Assessing prunability of network graph");
+
+						// The network graph must not be pruned while rapid sync completion is pending
+						let should_prune = match rapid_gossip_sync.as_ref(){
```

Comment:
Isn't it the case that user of `RapidGossipSync` won't also use `NetGraphMsgHandler`? If so, there would need to be mutually exclusive persistence paths. Otherwise, the graph will never be persisted.

Reply:
But if they don't opt to use the p2p updates (i.e., when `net_graph_msg_handler` is `None`), won't the `NetworkGraph` never be persisted?

Reply:
Yes, that is true. They do already have the downloaded snapshot, which actually takes less time to convert into a network graph than a conventionally serialized network graph, so I'm not sure that's really a problem.

Reply:
Should we not bother serializing the timestamp then? That way it can go inside `RapidGossipSync` instead of `NetworkGraph`. (cc: @TheBlueMatt)

Reply:
We should definitely bother serializing the timestamp. I think the real question is whether we should expose the NetworkGraph access from the RapidGossipSync struct for persistence purposes.

Reply:
> As for persisting the timestamp, I don't think we need to worry about persisting it or not, kinda who cares? And, indeed, someone may want to move from rapid to P2P or back, so might as well persist it.

Hmm... if it doesn't matter if we persist it, but including it in `NetworkGraph` so we can persist it is a hack, why do it? Seems to just add complexity while exposing an API that only some users would care about but largely only exists for the `lightning-rapid-gossip-sync` crate's internal usage.

Not persisting it but keeping it internal to `RapidGossipSync` only affects the restart case. Or would fetching new rapid sync data be so infrequent that it would only happen on a restart (i.e., user app is mostly closed)? And thus persisting the timestamp would be important.

Reply:
The main use case for rapid sync is mobile devices, where the most common state of the app will be "launched within the previous 30 seconds." Persisting the timestamp allows them to fetch differential snapshots whose size will be one or a couple hundred kB instead of 2MB. It can shave off a couple seconds on each app start.